src/raster/r.area.createweight/r.area.createweight.py

```diff
@@ -0,0 +1,1464 @@
+#%option G_OPT_V_MAP
+#% key: vector
+#% label: Vector with spatial units
+#% description: Polygon vector containing unique ID and response
+#               variable in the attribute table
+#% required : yes
+#% guisection: Required inputs
+#% guidependency:vector_layer,id,response_variable
+#%end
+#%option G_OPT_V_FIELD
+#% key: vector_layer
+#% label: Layer number or name
+#% required : yes
+#% guisection: Required inputs
+#% answer:1
+#% guidependency:id,response_variable
+#%end
+#%option G_OPT_DB_COLUMN
+#% key: id
+#% label: Name of the column containing unique ID of spatial units
+#% required : yes
+#% guisection: Required inputs
+#%end
+#%option G_OPT_DB_COLUMN
+#% key: response_variable
+#% label: Name of the column containing response variable
+#% description: Format: All values must be >0
+#% required : yes
+#% guisection: Required inputs
+#%end
+#%option G_OPT_R_INPUT
+#% key: basemap_a
+#% type: string
+#% label: Input raster 1
+#% description: E.g. Land cover, Land use, morphological areas...
+#% required : yes
+#% guisection: Required inputs
+#%end
+#%option G_OPT_R_INPUT
+#% key: basemap_b
+#% type: string
+#% label: Input raster 2 (optional)
+#% description: E.g. Land cover, Land use, morphological areas...
+#% required : no
+#% guisection: Optional inputs
+#%end
+#%option G_OPT_R_INPUT
+#% key: distance_to
+#% type: string
+#% label: Input distance raster (optional)
+#% description: Distance to zones of interest
+#% required : no
+#% guisection: Optional inputs
+#%end
+#%option
+#% key: tile_size
+#% key_desc: value
+#% type: Integer
+#% label: Spatial resolution of output weighting layer
+#% description: (in metres)
+#% required : yes
+#% guisection: Outputs
+#%end
+#%option G_OPT_R_OUTPUT
+#% key: output_weight
+#% description: Output weighting layer name
+#% required: yes
+#% guisection: Outputs
+#% answer:weighted_layer
+#%end
+#%option G_OPT_V_OUTPUT
+#% key: output_units
+#% description: Name for output vector gridded spatial units
+#% required: yes
+#% guisection: Outputs
+#% answer:gridded_spatial_units
+#%end
+#%option G_OPT_F_OUTPUT
+#% key: plot
+#% description: Name for output plot of model feature importances
+#% required: yes
+#% guisection: Outputs
+#%end
+#%option G_OPT_F_OUTPUT
+#% key: log_file
+#% description: Name for output file with log of the random forest run
+#% required: yes
+#% guisection: Outputs
+#%end
+#%option
+#% key: basemap_a_list
+#% type: string
+#% label: Categories of basemap A to be used
+#% description: Format: 1,2,3
+#% required: no
+#% guisection: Optional inputs
+#%end
+#%option
+#% key: basemap_b_list
+#% type: string
+#% label: Categories of basemap B to be used
+#% description: Format: 1,2,3
+#% required: no
+#% guisection: Optional inputs
+#%end
+#%flag
+#% key: a
+#% description: Use class names for basemap a
+#% guisection: Optional inputs
+#%end
+#%flag
+#% key: b
+#% description: Use class names for basemap b
+#% guisection: Optional inputs
+#%end
+#%option
+#% key: n_jobs
```

Comment:
We recently decided to add a standardized option for that. Not available in v7, but it will be in v8, so it depends what version you are aiming at. However, even for v7, you should still use the standardized name. See [Parser standard options](https://grass.osgeo.org/grass80/manuals/parser_standard_options.html):

| standard option | description | key |
| --- | --- | --- |
| G_OPT_M_NPROCS | Number of threads for parallel computing | nprocs |

In any case, please give your feedback on the name (key) and description. Ideally as an issue. Now would still be time to change it.

Reply:
It is done. The key and description is clear to me ;)